frontend/src/modules/machinery/components/Translation.js

```diff
@@ -79,39 +81,52 @@ export default function Translation(props: Props) {
                 onClick={copyTranslationIntoEditor}
                 ref={translationRef}
             >
-                <header>
-                    {!translation.quality ? null : (
-                        <span className='quality'>
-                            {translation.quality + '%'}
-                        </span>
-                    )}
-                    <TranslationSource
+                {translation.sources.includes('concordance-search') ? (
+                    <ConcordanceSearch
+                        sourceString={sourceString}
                         translation={translation}
-                        locale={locale}
                     />
-                </header>
-                <p className='original'>
-                    {translation.sources.indexOf('caighdean') === -1 ? (
-                        <GenericTranslation
-                            content={translation.original}
-                            diffTarget={sourceString}
-                        />
-                    ) : (
-                        /*
-                         * Caighdean takes `gd` translations as input, so we shouldn't
-                         * diff it against the `en-US` source string.
-                         */
-                        <GenericTranslation content={translation.original} />
-                    )}
-                </p>
-                <p
-                    className='suggestion'
-                    dir={locale.direction}
-                    data-script={locale.script}
-                    lang={locale.code}
-                >
-                    <GenericTranslation content={translation.translation} />
-                </p>
+                ) : (
+                    <>
+                        <header>
+                            {!translation.quality ? null : (
```

Comment:
Can `translation.quality` be `0`? If not, then `translation.quality && <span...>` should suffice, i.e. no need to check for the negative condition.

Reply:
We use the `!condition ? null : component` pattern a lot in our code. I don't have strong opinions about the code style, and I also find the proposed one cleaner. I appreciate consistency, though.

So if you go ahead with the proposed change, please also file a bug to replicate the change on other instances of the pattern.

Reply:
Yes, the formatting here was in keeping with the existing pattern within the code. However, I also prefer the `&&` format and use it extensively in my other work (in fact, I realized that I used it when displaying the button in this MR without thinking 😁 ).

I will go forward with the suggested format and file the bug to go back and replace the existing instances as I agree that consistency is important.